datasync/chaindatafetcher/kas/repository_transactions.go

```diff
@@ -0,0 +1,183 @@
+package kas
+
+import (
+	"fmt"
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/common/hexutil"
+	"strings"
+)
+
+// TxFilteringTypes filters types which are only stored in KAS database.
+var TxFilteringTypes = map[types.TxType]bool{
+	types.TxTypeLegacyTransaction: true,
+
+	types.TxTypeValueTransfer:                      true,
+	types.TxTypeFeeDelegatedValueTransfer:          true,
+	types.TxTypeFeeDelegatedValueTransferWithRatio: true,
+
+	types.TxTypeValueTransferMemo:                      true,
+	types.TxTypeFeeDelegatedValueTransferMemo:          true,
+	types.TxTypeFeeDelegatedValueTransferMemoWithRatio: true,
+
+	types.TxTypeSmartContractDeploy:                      true,
+	types.TxTypeFeeDelegatedSmartContractDeploy:          true,
+	types.TxTypeFeeDelegatedSmartContractDeployWithRatio: true,
+
+	types.TxTypeSmartContractExecution:                      true,
+	types.TxTypeFeeDelegatedSmartContractExecution:          true,
+	types.TxTypeFeeDelegatedSmartContractExecutionWithRatio: true,
+}
+
+// transformToKasTxs transforms a chain event to transaction list according to the KAS database scheme.
+func transformToKasTxs(event blockchain.ChainEvent) []*Tx {
+	var txs []*Tx
+	// TODO-ChainDataFetcher extract EOAs in order to expire the KAS cache
+	//expiredEOAs := make(map[string]struct{})
+
+	block := event.Block
+	head := block.Header()
+	receipts := event.Receipts
+
+	for idx, rawTx := range block.Transactions() {
+		txId := head.Number.Int64()*maxTransactionCount*maxTransactionLogCount + int64(idx)*maxInternalTransactionCount
+
+		// from
+		var from common.Address
+		if rawTx.IsLegacyTransaction() {
+			signer := types.NewEIP155Signer(rawTx.ChainId())
+			from, _ = types.Sender(signer, rawTx)
+		} else {
+			from, _ = rawTx.From()
+		}
+
+		// to
+		to := rawTx.To()
+		if to == nil {
+			to = &common.Address{}
+		}
+
+		// value
+		value := hexutil.EncodeBig(rawTx.Value())
+
+		// transaction hash
```

Comment:
no process?

Reply:
No process. Removed the comment d0b0c6cb753b422717a9f7388278c6d73a38fe17